drivers/syslog/syslog_filechannel.c

```diff
@@ -85,28 +88,44 @@ static void log_rotate(FAR const char *log_file)
       return;
     }
 
-  /* Construct the backup file name. */
+  /* Rotated file names. */
 
-  backup_file = malloc(strlen(log_file) + 3);
-  if (backup_file == NULL)
+  name_size = strlen(log_file) + 8;
+  rotate_to = malloc(name_size);
+  rotate_from = malloc(name_size);
+  if ((rotate_to == NULL) || (rotate_from == NULL))
     {
-      return;
+      goto end;
     }
 
-  sprintf(backup_file, "%s.0", log_file);
+  /* Rotate the logs. */
+
+  for (int i = (CONFIG_SYSLOG_FILE_ROTATIONS - 1); i > 0; i--)
```

Comment:
Please declare "int i" at the top of the function, NuttX common code needs to be C89 compatible, only Arch specific code doesn't need to follow it.

Reply:
done!

Reply:
Thank you!
BTW, I think it should be i = CONFIG_SYSLOG_FILE_ROTATIONS because by default CONFIG_SYSLOG_FILE_ROTATIONS = 1, so 1 - 1 = 0 => the "i > 0" test will fail and it will not enter inside the "for()"

Reply:
Sorry, I missed the remove and rename after the for(), it is right!